### PR TITLE
refactor: Remove vestiges of python2 support

### LINKED ISSUE #677: Remove usage of `six`
`six` and the things it provides are only needed for the transition from python2 to python3. As it seems that this library no longer supports python2, it'd be nice to remove the usage of this library.

Happy to help clean it up

### PR DESCRIPTION
**Pull request by a-ungurianu**
Fixes #677 

## Why is this needed?

Not "needed", but this removes unnecessary dependencies on `six` and `selectors2` making this package a no-deps package for runtime usage. 

More of a spring cleaning

## Proposed Changes

  - remove usage of `six`
  - remove ImportError conditional statements that were due to `python3` stdlib updates

## Does this PR introduce any breaking change?

Should be completely backwards compatible. Exceptions might change a bit as we use `python3`'s `raise ... from ...` syntax

### CODE REVIEW

File: kazoo/handlers/threading.py
Comment (ceache) [inline kazoo/handlers/threading.py:24]: I would prefer you rename it `queue` instead of keeping the legacy capitalized name

File: kazoo/handlers/utils.py
Comment (ceache) [inline kazoo/handlers/utils.py:20]: `fcntl` is not a py27 thing (it was present then too). It is a runtime thing.

File: kazoo/handlers/utils.py
Comment (ceache) [inline kazoo/handlers/utils.py:356]: [nit] `isinstance()` accepts a list as second argument

**Review by ceache — COMMENTED** [review]
Looks good. Thanks for taking care of this

File: kazoo/handlers/utils.py
Comment (a-ungurianu) [reply kazoo/handlers/utils.py:20]: Interesting. The docs says that it's not supported only in WASM. Are there people who use this lib with WASM?
I agree with keeping it, just curious

Comment (ceache) [reply kazoo/handlers/utils.py:20]: Not that I know of....
It may simply be that much older versions of python did not have it.

File: kazoo/protocol/connection.py
Comment (jeffwidman) [inline kazoo/protocol/connection.py:809]: Do we still need the `: %s" % err` in these three?

Would `raise SASLException("library error") from err` work?

File: kazoo/protocol/connection.py
Comment (jeffwidman) [inline kazoo/protocol/connection.py:829]: This is a bit inconsistent with the refactoring above that adds a `% err` to the error messages, but here is omitted.

I suspect this is the right format and the others should be updated, but if not then this should be tweaked for consistency.

**Comment by jeffwidman** [comment]
Thanks for tackling this!

File: kazoo/protocol/connection.py
Comment (ceache) [reply kazoo/protocol/connection.py:829]: [nit] I would also recommend you keep using `err` for exceptions, to stay consistent with the others.

File: kazoo/recipe/lock.py
Comment (ceache) [inline kazoo/recipe/lock.py:203]: [nit] ditto about the consistency of using `err` vs `ex`.

**Review by ceache — COMMENTED** [review]
Couple minor nitpicks. LGTM otherwise

File: kazoo/protocol/connection.py
Comment (a-ungurianu) [reply kazoo/protocol/connection.py:809]:
```
>>> def throw():
...     raise KeyError("Message 1")
...
>>> def throw_through():
...     try:
...         throw()
...     except KeyError as err:
...         raise Exception("Message 2") from err
...
>>> throw_through()
Traceback (most recent call last):
  File "<stdin>", line 3, in throw_through
  File "<stdin>", line 2, in throw
KeyError: 'Message 1'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 5, in throw_through
Exception: Message 2
```

This is what this would produce. Happy to drop it, just want to make sure we're happy with the "change" in exception messaging

File: kazoo/recipe/lock.py
Comment (a-ungurianu) [reply kazoo/recipe/lock.py:193]: There's not a lot of consistency overall. I was inferring the name from the class name, `...Exception as ex`, `...Error as err`.

What do you think?

Comment (ceache) [reply kazoo/recipe/lock.py:193]: I think that makes perfect sense 😄 

I did a quick poll of the codebase and it is not very consistent indeed.

We currently mostly have `err`, `exc` and (my "favorite") the ambivalent `e`.

   1 ex
   6 err
   6 exc
   7 e

I would suggest we then follow your model but with `*Error` / `*Exception` aligned with `err` / `exc` respectively.

We can remediate the others overtime.
Would you be OK with that?

File: kazoo/protocol/connection.py
Comment (ceache) [reply kazoo/protocol/connection.py]: I think this is OK.  Having the stringification of parent exception in the message does not provide any value.

We only expose "library", "protocol" and "unknown" messages at this point anyway. I think this is good 👍

File: kazoo/tests/conftest.py
Comment (StephenSorriaux) [inline kazoo/tests/conftest.py:13]: Some tests does not make use of the cluster, I think the `else` case can just be omitted here. What do you think?

File: kazoo/recipe/lock.py
Comment (StephenSorriaux) [inline kazoo/recipe/lock.py:198]: Can't we just `raise` here since it seems we are not making use of `exc`?

Comment (a-ungurianu) [reply kazoo/recipe/lock.py:198]: That's very fair. Forgot you can do that.

File: kazoo/tests/conftest.py
Comment (a-ungurianu) [reply kazoo/tests/conftest.py:13]: I think you're right. The lack of a cluster attribute is a marker that the test was not meant to have a cluster.

I'll remove the else.

**Review by StephenSorriaux — APPROVED** [review]
Many thanks for your work on this!

**Comment by StephenSorriaux** [comment]
Gentle ping @ceache and @jeffwidman, would you like to give it a last look?

File: kazoo/handlers/eventlet.py
Comment (jeffwidman) [inline kazoo/handlers/eventlet.py:5]: No need for blank line, these are all stdlib imports, right? So can be grouped together in lexicographic order. This comment applies multiple places in the PR.

File: kazoo/handlers/threading.py
Comment (jeffwidman) [inline kazoo/handlers/threading.py:25]: Shouldn't this be grouped with the `stdlib` imports group above?

File: kazoo/protocol/connection.py
Comment (jeffwidman) [inline kazoo/protocol/connection.py:813]: I know this was already partially discussed, but why mix `err` and `exc`? Here it just adds confusion... pick one in this section for consistency...

File: kazoo/recipe/lock.py
Comment (jeffwidman) [inline kazoo/recipe/lock.py:18]: Why the rename? Why not `import monotonic` and use `monotonic` at the callsite? The original rename was because of the try/catch import, but here that's no longer relevant...

**Review by jeffwidman — COMMENTED** [review]
A few minor nits. I may be off on them, and I'd also be okay if this shipped as-is, but if you want my $0.02 here it is 😀

File: kazoo/recipe/lock.py
Comment (ceache) [reply kazoo/recipe/lock.py:18]: I would further say that I much prefer importing modules instead of classes or functions (kind of like one would use namespaces in C++).

This is much more manageable because you avoid any name collision and it is very clear where each symbol comes from.

``` python
import time

def foo():
    tick = time.monotonic()

```

**Review by ceache — COMMENTED** [review]
I have done one last pass.
Code is good, this is all styling feedback.

Thanks for all this!

**Comment by a-ungurianu** [comment]
Any idea why isort take care of the import ordering/grouping?

File: kazoo/handlers/gevent.py
Comment (a-ungurianu) [reply kazoo/handlers/gevent.py:21]: I can tag it along as a separate commit

**Review by StephenSorriaux — APPROVED** [review]

**Comment by StephenSorriaux** [comment]
@a-ungurianu Thank you again for this. I'll let you free to merge this whenever you want, and, if I am not mistaken, make it your first merge as an official Kazoo contributor :rocket:

**Review by ceache — APPROVED** [review]
Thanks for all of this 👍

**Comment by a-ungurianu** [comment]
Didn't realize I can press the big green button :). Here goes nothing

**Comment by jeffwidman** [comment]
Thanks again for driving this to completion! Congrats on your first merge.

**Pull request merged**
